block-layer/codecs/dag-cbor.md

```diff
@@ -37,93 +39,103 @@ In DAG-CBOR, map keys must be strings, as defined by the [IPLD Data Model]. Othe
 
 ## Implementations
 
 ### JavaScript
 
-[dag-cbor], used by [ipld] and [@ipld/block] adheres to this specification, with the following caveats:
+**[@ipld/dag-cbor]**, for use with [multiformats] adheres to this specification, with the following caveats:
+ * Complete strictness is not yet enforced on decode. Specifically: correct map ordering is not enforced and floats that are not encoded as 64-bit are not rejected.
+ * [`BigInt`] is accepted along with `Number` for encode, but the smallest-possible rule is followed when encoding. When decoding integers outside of the JavaScript "safe integer" range, a [`BigInt`] will be used.
+
+The legacy **[ipld-dag-cbor]** implementation adheres to this specification, with the following caveats:
 
- * Strictness is not yet enforced on decode, blocks encoded that don't follow the strictness rules are not rejected
- * Floating point values are encoded as their smallest form rather than always double-precision.
- * Many additional object types outside of the Data Model are currently accepted for encoding.
- * IEEE 754 special values `NaN`, `Infinity` and `-Infinity` are accepted for decode and encode.
+ * Strictness is not enforced on decode; blocks encoded that do not follow the strictness rules are not rejected.
+ * Floating point values are encoded as their smallest form rather than always 64-bit.
+ * Many additional object types outside of the Data Model are currently accepted for decode and encode, including `undefined`.
+ * [IEEE 754] special values `NaN`, `Infinity` and `-Infinity` are accepted for decode and encode.
+ * Integers outside of the JavaScript "safe integer" range will use the third-party [bignumber.js] library to represent their values.
+
+Note that inability to clearly differentiate between integers and floats in JavaScript may cause problems with round-trips of floating point values. See the [IPLD Data Model] and the discussion on [Limitations](#limitations) below for further discussion on JavaScript numbers and recommendations regarding the use of floats.
 
 ### Go
 
-[ipld-cbor] and [ipld-prime] adhere to this specification, with the following caveats:
+**[go-ipld-cbor]** and **[go-ipld-prime]** adhere to this specification, with the following caveats:
 
- * Strictness is not yet enforced on decode, blocks encoded that don't follow the strictness rules are not rejected
- * IEEE 754 special values `NaN`, `Infinity` and `-Infinity` are accepted for decode and encode.
+ * Strictness is not enforced on decode; blocks encoded that do not follow the strictness rules are not rejected.
+ * [IEEE 754] special values `NaN`, `Infinity` and `-Infinity` are accepted for decode and encode.
```

Comment:
same question, LGTM otherwise

Reply:
I think we should leave statements about when/how strictness is applied. The spec should define the strictness and implementations that want to violate the strictness can document when and where they do so.

We should strive for an equilibrium in which strictness is assumed to be the norm and is explicitly called out when not applied.

Reply:
> strictness is assumed to be the norm and is explicitly called out when not applied

that's what this section is doing, are you suggesting I remove this?

Reply:
It seems we should probably remove (or at least put behind a flag) our support for these "numbers", yes.  (I have no idea what our code currently does with these, in detail.)

Reply:
Sorry, I'm getting mixed signals in this thread. For clarity - refmt supports `NaN`, `Infinity` and `-Infinity` but I'm assuming this will change at some point in the future, perhaps when ipld-prime absorbs more refmt functionality and does its own CBOR. I don't view this as particularly urgent work and don't think these listed items should be viewed as marks against the libraries, just informational for users.

@mikeal @warpfork are you suggesting I remove comments about how the various libraries conform or don't conform from here? That would seem to be a negative for this document, they're informational for users. If you have suggestions for how to reframe the wording then let me know.

Reply:
I am happy about how this is currently phrased.  (Github's UI may have caused me and Mikeal tunnel vision earlier; some views "helpfully" cut off the line introducing this list stating that it's a list of exceptions to the rule.  I think we're all in agreement.)